haystack_experimental/components/converters/pdf_to_image.py

```diff
@@ -0,0 +1,140 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+
+from haystack import component, logging
+from haystack.components.converters.utils import get_bytestream_from_source, normalize_metadata
+from haystack.dataclasses import ByteStream
+from haystack.utils import expand_page_range
+
+from haystack_experimental.components.converters.image_utils import (
+    convert_pdf_to_images,
+    pypdf_and_pdf2image_import,
+)
+from haystack_experimental.dataclasses.chat_message import ImageContent
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class PDFToImageContent:
+    """
+    Converts PDF files to ImageContent objects.
+    """
+
+    def __init__(
+        self,
+        *,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        size: Optional[Tuple[int, int]] = None,
+        page_range: Optional[List[Union[str, int]]] = None,
+    ):
+        """
+        Create the PDFToImageContent component.
+
+        :param detail: Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+            This will be passed to the created ImageContent objects.
+        :param size: If provided, resizes the image to fit within the specified dimensions (width, height) while
+            maintaining aspect ratio. This reduces file size, memory usage, and processing time, which is beneficial
+            when working with models that have resolution constraints or when transmitting images to remote services.
+        :param page_range: List of page numbers and/or page ranges to convert to images. Page numbers start at 1.
+            If None, all pages in the PDF will be converted. Pages outside the valid range (1 to number of pages)
+            will be skipped with a warning. For example, page_range=[1, 3] will convert only the first and third
+            pages of the document. It also accepts printable range strings, e.g.:  ['1-3', '5', '8', '10-12']
+            will convert pages 1, 2, 3, 5, 8, 10, 11, 12.
+        """
+        self.detail = detail
+        self.size = size
+        self.page_range = page_range
+        pypdf_and_pdf2image_import.check()
+
```

Comment:
do we also need `pillow_import.check()` here?

Reply:
No because pdf2image will download PIL when installing pdf2image.